src/components/shared/passphraseRenderer/index.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import grid from 'flexboxgrid/dist/flexboxgrid.css';
+import { withTranslation } from 'react-i18next';
+import styles from './passphraseRenderer.css';
+
+class PassphraseRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      fieldSelected: undefined,
+      displayOptions: undefined,
+      chosenWords: {},
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.hasErrors && !this.props.hasErrors) {
+      this.setState({
+        fieldSelected: undefined,
+        displayOptions: undefined,
+        chosenWords: {},
+      });
+    }
+  }
+
+  // eslint-disable-next-line complexity
+  getStyle(i) {
+    const {
+      missingWords, isConfirmation, hasErrors, isCorrect,
+    } = this.props;
+    const { fieldSelected, chosenWords } = this.state;
+
+    if (!missingWords) return styles.default;
+    if (chosenWords[i] && hasErrors) return styles.error;
+    if (chosenWords[i] && isCorrect) return styles.correct;
+    if (chosenWords[i]) return styles.selected;
+    if (missingWords.includes(i) && fieldSelected === i) return styles.emptyInputFocused;
+    if (missingWords.includes(i) && fieldSelected !== i) return styles.emptyInput;
+    if (isConfirmation) return styles.disabled;
+    return styles.default;
+  }
```

Comment:
we need to improve this function to remove eslint complexity.

Reply:
I have moved some of that logic to other functions. Let me know if that looks ok to you.